src/shady/parser/token.c

```diff
@@ -115,13 +115,15 @@ Token next_token(Tokenizer* tokenizer) {
                 token_size++;
             } else break;
         }
-    } else if (is_digit(slice[0])) {
+    } else if (is_digit(slice[0]) || (slice[0] == '-') && in_bounds(tokenizer, token_size + 1) && is_digit(slice[1])) {
```

Comment:
I reverted this because it broke infix `-` without whitespace. See https://github.com/Hugobros3/shady/commit/4e25f07ca23b1047c1fcc1ea59c97f6f7b57e552